doc/classes/ImageTexture.xml

```diff
@@ -4,10 +4,36 @@
 		A [Texture2D] based on an [Image].
 	</brief_description>
 	<description>
-		A [Texture2D] based on an [Image]. Can be created from an [Image] with [method create_from_image].
-		[b]Note:[/b] The maximum image size is 16384×16384 pixels due to graphics hardware limitations. Larger images will fail to import.
+		A [Texture2D] based on an [Image]. For an [Image] to be displayed, an [ImageTexture] has to be created from it using the [method create_from_image] method:
+		[codeblock]
+		var texture = ImageTexture.new()
+		var image = Image.new()
+		image.load("res://icon.png")
+		texture.create_from_image(image)
+		$Sprite.texture = texture
+		[/codeblock]
+		[b]Warning:[/b] Always prefer to load imported textures over loading them from within the filesystem dynamically, as it may not work in exported projects:
```

Comment:
I've seen this warning multiple times from many sources, but still don't understand the why's of it. I also continue to not understand how to know when an texture is "imported" properly v.s. just present. There is also an implication that "imported" objects are inside the project folder before the game is exported but I've not seen this clearly defined. Lastly what should someone do if they truly intend for an image/texture to be loaded from `user://`?

Reply:
In theory this should be documented at [`Import process`](https://docs.godotengine.org/en/stable/getting_started/workflow/assets/import_process.html) page describing the reasons why assets like images have to be imported in Godot.

By default, any foreign file format is not recognized by Godot, so it will be ignored and not visible throughout the "FileSystem" dock (which looks into `res://` path, which happens to be a project's root on your filesystem, or the contents of `game.pck` when you export a project).

On the high-level, files like `*.png` are recognized by Godot, but they have to be made compatible with the existing datatypes first, like `Image` or `Texture`. Therefore, those foreign files have to be converted first, aka imported as `Resource`s. Those imported resources get saved into the `.import` folder which is hidden in Godot's UI.

The problem is that there may be different ways to convert such files, so the "Import" dock gives you the ability to configure the default import procedure for each file, as described in https://github.com/godotengine/godot/issues/20972#issuecomment-645514487.

Now, if you want to actually interact with those imported resources, loading them by the "original path" is enough:

```gdscript
var texture = load("res://icon.png")
```

What happens is that the `ResourceLoader` takes the original path ("res://icon.png"), looks whether such a file was previously imported (looks into `.import` directory), and retrieves the resource for you, so it doesn't really load the original file anymore.

Now, the problem with such approach is that this is limited to the project's virtual resource filesystem ("res://"). If you need to load any external files outside of "res://" in exported projects, this will fail, because all of the game contents get packed into `game.pck`. Methods like `Image.load()` are not even aware of this virtual filesystem in Godot and rely on the typical OS-specific filesystem methods for loading data (using the internal `FileAccess` class, aka `File` in GDScript), and not GDScript's `load()`. The reason it works in the editor is because those image files are still there on your filesystem. In theory, you can also disable automatic re-import, delete those original files outside of the editor, and still able to run your game. 🙂

> Lastly what should someone do if they truly intend for an image/texture to be loaded from `user://`?

For best results, you have to replicate the same import process manually at run-time, because most of the import methods are only available in the editor. Yes, I know, this can be a big limitation in your project if your game takes advantage of user-generated content. And the most practical and feasible way to "import" textures at run-time is by loading an `Image` and creating an `ImageTexture` from that. 🙂 

The procedure would be the same as in the first snippet in this PR, but use "user://" instead. You have to clearly define what content would be able to be loaded externally and design your game architecture accordingly.

Reply:
thanks this was very helpful